File: Java Fundamentals-Jan2020/src/b_BasicSyntaxCondStatementsLoops_Exercise/PadawanEquipment.java
package b_BasicSyntaxCondStatementsLoops_Exercise;

import java.util.Scanner;

public class PadawanEquipment {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double money = Double.parseDouble(sc.nextLine());       //0.00…1,000.00
        int students = Integer.parseInt(sc.nextLine());         //0…100
        double saberPrice = Double.parseDouble(sc.nextLine());  //0.00…100.00
        double robePrice = Double.parseDouble(sc.nextLine());   //0.00…100.00
        double beltPrice = Double.parseDouble(sc.nextLine());   //0.00…100.00

        int beltFree = students / 6;

        double saberCost = Math.ceil(1.1 * students) * saberPrice;
        double robeCost = students * robePrice;
        double beltCost = (students - beltFree) * beltPrice;

        double totalCost = saberCost + robeCost + beltCost;

        if (money >= totalCost) {
            System.out.println(String.format("The money is enough - it would cost %.2flv.", totalCost));
        } else {
            System.out.println(String.format("Ivan Cho will need %.2flv more.", (totalCost - money)));
        }
    }
}

File: CSharp-Fundamentals-2023/01-1.ConditStatementsAndLoops-Exercise/09.PadawanEquipment/PadawanEquipment.cs
/*
 * Yoda is starting his newly created Jedi academy. So, he asked master John to buy the needed equipment. 
 * The number of items depends on how many students will sign up. The equipment for each Padawan contains: Lightsaber, Belt and Robe
 * 
 * You will be given the amount of money John has, the number of students and the prices of each item. 
 * Calculate if John has enough money to buy equipment for each Padawan or how much more money he needs.
 * 
 * There are some additional requirements:
 *  - Lightsabres sometimes break, so John should buy 10% more (taken from the students' count), rounded up to the next integer.
 *  - Every sixth belt is free.
 */


//INPUT
double money = double.Parse(Console.ReadLine());        //[0.00…1000.00]
int studentsCount = int.Parse(Console.ReadLine());      //[0…100]
double saberPrice = double.Parse(Console.ReadLine());   //[0.00…100.00]
double robePrice = double.Parse(Console.ReadLine());    //[0.00…100.00]
double beltPrice = double.Parse(Console.ReadLine());    //[0.00…100.00]

//LOGIC
double costForSabers = Math.Ceiling(studentsCount * 1.1) * saberPrice;
double costForRobes = studentsCount * robePrice;

int freeBeltsCount = studentsCount / 6;
double costForBelts = (studentsCount - freeBeltsCount) * beltPrice;

double totalCost = costForSabers + costForRobes + costForBelts;

//OUTPUT
if (money >= totalCost)
{
    Console.WriteLine($"The money is enough - it would cost {totalCost:f2}lv.");
}
else
{
    Console.WriteLine($"John will need {(totalCost - money):f2}lv more.");
}